src/app/archive/archive_lib/processor.ml

```diff
@@ -863,27 +879,49 @@ module Block_and_signed_command = struct
           , fee_payer_account_creation_fee_paid
           , receiver_account_creation_fee_paid
           , created_token
-          , balances_to_int64s balances )
+          , balances )
       | Failed (failure, balances) ->
-          ( "failed"
-          , Some failure
-          , None
-          , None
-          , None
-          , balances_to_int64s balances )
+          ("failed", Some failure, None, None, None, balances)
+    in
+    let add_optional_balance id balance =
+      match balance with
+      | None ->
+          Deferred.Result.return None
+      | Some balance ->
+          let%map balance_id =
+            Balance.add_if_doesn't_exist
+              (module Conn)
+              ~public_key_id:id ~balance
+          in
+          Some balance_id
+    in
+    let%bind fee_payer_balance_id =
+      Balance.add_if_doesn't_exist
+        (module Conn)
+        ~public_key_id:fee_payer_id
+        ~balance:
+          (Option.value_exn
+             fee_payer_balance
+             (* TODO(omerzach): What should we do when this is None? *))
```

Comment:
@mrmr1993 do you know what I need to do here?

Reply:
What you've done here is fine. Any transaction included in a block will have had its fee paid, and so the fee payer balance will always have been updated.

At some point I will update the underlying type to actually reflect this reality, but there's a bit of complexity because we need to keep the same derived JSON encoding. For now, we just have to pretend :)

It would be nice if you could add words to this effect in a comment here, so that the `value_exn` doesn't look as scary.

Reply:
Okay great, will replace the TODO with that explanation then.